tests/test_raster_manipulation.py

```diff
@@ -0,0 +1,48 @@
+import xarray as xr
+import pandas as pd
+from src.utils_general.raster_manipulation import (
+    compute_raster_statistics,
+)
+from shapely.geometry import Polygon
+from pandas._testing import assert_frame_equal
+import geopandas as gpd
+
+
+def test_compute_raster_statistics():
+    da = xr.DataArray(
+        [[1, 2, 3], [4, 5, 6]],
+        dims=("y", "x"),
+        coords={"y": [1.5, 0.5], "x": [0.5, 1.5, 2.5]},
+    ).rio.write_crs("EPSG:4326", inplace=True)
+
+    d = {
+        "name": ["hi", "bye"],
+        "geometry": [
+            Polygon([(0, 0), (0, 2), (2, 2), (2, 0)]),
+            Polygon([(2, 0), (2, 2), (3, 2), (3, 0)]),
+        ],
+    }
+    gdf = gpd.GeoDataFrame(d)
+
+    df_stats = compute_raster_statistics(
+        gdf, "name", da, stats_list=["min", "max", "mean", "count", "sum"]
```

Comment:
Could you also throw in a couple of percentiles to test ;)

Reply:
Good one, done. With this small number of data points the quantiles are a bit funny, as some linear interpolation is being done. But I do think that is the most valide method (you could also e.g. take the nearest or max value without interpolation)